### PR TITLE
Feature/icky 305

### PR DESCRIPTION
**Pull request by kibo-susantasamanta**
@kibo-kevinwatts @kibo-sushant @Chandradeepta

### CODE REVIEW

File: composables/getters/productGetters.ts [hunk 1]
```diff
@@ -65,6 +70,33 @@ export const getOptionSelectedValue = (option: ProductOption) => {
 }
 export const getOptionName = (option: ProductOption): string => option?.attributeDetail?.name || ""
 export const getOptions = (product: Product) => product?.options
+export const getSearchSuggestions = (suggestionSearch: SearchSuggestionResult) => {
```
Comment (kibo-kevinwatts) [inline composables/getters/productGetters.ts]: move getSearchSuggestions to its own getter file

File: components/KiboAppHeader.vue [hunk 2]
```diff
@@ -106,15 +149,69 @@ export default defineComponent({
...
+      } else {
+        term.value = ""
+        return searchBarRef.value.$el.children[0].focus()
+      }
+    }
+    const gotoSearchResult = () => {
+      if (term.value) {
+        const searchStr = term.value
+        closeSearch()
+        return app.router.push({ path: "/search", query: { searchStr } })
```
Comment (kibo-kevinwatts) [inline components/KiboAppHeader.vue]: query: { phrase: searchStr}

File: nuxt.config.ts [hunk 3]
```diff
@@ -147,6 +147,11 @@ export default defineNuxtConfig({
         path: "/c/:categoryCode",
         component: resolve(__dirname, "pages/category/index.vue"),
       })
+      routes.push({
+        name: "search_details",
+        path: "/search/:searchStr",
```
Comment (kibo-kevinwatts) [inline nuxt.config.ts]: remove :searchStr

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: composables/getters/productGetters.ts (hunk 1 above)
Comment (kibo-susantasamanta) [reply composables/getters/productGetters.ts]: Resolved. created searchSuggestiongetters for that.

File: nuxt.config.ts (hunk 3 above)
Comment (kibo-susantasamanta) [reply nuxt.config.ts]: Removed.

File: components/KiboAppHeader.vue (hunk 2 above)
Comment (kibo-susantasamanta) [reply components/KiboAppHeader.vue]: Resolved.

**Review by kibo-kevinwatts — APPROVED** [review]

File: components/KiboAppHeader.vue [hunk 4]
```diff
@@ -110,8 +118,10 @@ import { useAsync } from "@nuxtjs/composition-api"
 import useUiState from "../composables/useUiState"
 import * as logo from "../assets/kibo_logo.png"
 import { usePurchaseLocation } from "../composables"
```
Comment (kibo-sushant) [inline components/KiboAppHeader.vue]: Instead of using .../  use @/composables for consistency purpose

File: components/KiboAppHeader.vue [hunk 5]
```diff
@@ -156,17 +168,17 @@ export default defineComponent({
...
-      isSearchOpen.value = true
-      // @todo use result to show it in search suggestion dialog
+      searchSuggestionResult.value = {}
+      try {
+        isSearchOpen.value = true
+        if (!paramValue.target) {
+          term.value = paramValue
+        } else {
+          term.value = paramValue.target.value
+        }
```
Comment (kibo-sushant) [inline components/KiboAppHeader.vue]: simplyfy something like below 
term.value = paramValue.target ? paramValue.target.value : paramValue

File: components/KiboSearchBar.vue [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    <slot name="icon">
+      <SfButton class="sf-search-bar__button sf-button--pure" @click="$emit('click', value)">
+        <span v-if="icon" class="sf-search-bar__icon">
+          <SfIcon :color="icon.color" :size="icon.size" icon="search" />
+        </span>
+      </SfButton>
+    </slot>
+  </div>
+</template>
+<script>
```
Comment (kibo-sushant) [inline components/KiboSearchBar.vue]: use lang

File: components/KiboSearchBar.vue [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    },
+    /**
+     * Value that will be displayed in search bar
+     */
+    value: {
+      type: [Number, String],
+      default: null,
+    },
+    /**
+     * Object to define icon look. Should have values for color and size
```
Comment (kibo-sushant) [inline components/KiboSearchBar.vue]: use oneliner comments everywere

File: components/KiboSearchBar.vue [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        input: (event) => this.$emit("input", event.target.value),
+        "keyup.enter": (event) => this.$emit("input", event.target.value),
+        "keyup.esc": () => this.$emit("input", ""),
+        blur: () => this.$emit("blur"),
+      }
+    },
+  },
+}
+</script>
+<style lang="scss">
```
Comment (kibo-sushant) [inline components/KiboSearchBar.vue]: Apply scoped style if possible

File: components/KiboSearchSuggestion.vue [hunk 9]
```diff
@@ -0,0 +1,217 @@
+<template>
+  <div v-show="visible" class="custom-search-header">
+    <div class="search-container">
+      <div class="search-product left">
+        <SfLoader v-if="loading" :class="{ loader: loading }" :loading="loading">
+          <div>{{ $t("Searching......") }}</div>
```
Comment (kibo-sushant) [inline components/KiboSearchSuggestion.vue:6]: Avoid hardcoding in template

File: components/KiboSearchSuggestion.vue [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+          <div v-if="result.products && result.products.length > 0" class="product-card-list">
+            <template v-for="(product, index) in result.products">
+              <KiboProductCard
+                :key="index"
+                class="product_card_details"
+                :image="product.productImageUrls[0]"
+                :title="product.productName"
+                image-width="97px"
+                image-height="97px"
+                :link="localePath(`/product/${product.productCode}`)"
```
Comment (kibo-sushant) [inline components/KiboSearchSuggestion.vue:19]: Avoid hard coding height and width in pixel

File: components/KiboSearchSuggestion.vue [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+          </div>
+          <div v-else class="no-record">
+            <div v-if="!loading">No Result found......</div>
+          </div>
+        </div>
+      </div>
+      <div class="vertical-line"></div>
+      <div class="search-category right">
+        <SfLoader v-if="loading" :class="{ loader: loading }" :loading="loading">
+          <div>{{ $t("Searching......") }}</div>
```
Comment (kibo-sushant) [inline components/KiboSearchSuggestion.vue:33]: Avoid hardcoding text here

File: components/KiboSearchSuggestion.vue [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+      getCatLink,
+      closeSearchProductClick,
+      gotToCategory,
+      props,
+    }
+  },
+})
+</script>
+
+<style lang="scss">
```
Comment (kibo-sushant) [inline components/KiboSearchSuggestion.vue:117]: use Scoped style if possible

File: composables/useSearchSuggestions.ts [hunk 13]
```diff
@@ -21,8 +21,10 @@ export const useSearchSuggestions = () => {
         })
         result.value = response.data.suggestionSearch
       }
-    } catch (err) {}
-    loading.value = false
+    } catch (err) {
```
Comment (kibo-sushant) [inline composables/useSearchSuggestions.ts:24]: console the error

File: composables/useSearchSuggestions.ts [hunk 14]
```diff
@@ -21,8 +21,10 @@ export const useSearchSuggestions = () => {
         })
         result.value = response.data.suggestionSearch
```
Comment (kibo-sushant) [inline composables/useSearchSuggestions.ts]: use response?.data?.suggestionSearch instead

File: components/KiboAppHeader.vue (hunk 4 above)
Comment (kibo-susantasamanta) [reply components/KiboAppHeader.vue]: solved.

File: components/KiboAppHeader.vue (hunk 5 above)
Comment (kibo-susantasamanta) [reply components/KiboAppHeader.vue]: done.

File: components/KiboSearchBar.vue (hunk 7 above)
Comment (kibo-susantasamanta) [reply components/KiboSearchBar.vue]: Done

File: components/KiboSearchSuggestion.vue (hunk 11 above)
Comment (kibo-susantasamanta) [reply components/KiboSearchSuggestion.vue:33]: as per VSF it's ok. no need to add extra variable.

File: components/KiboSearchSuggestion.vue (hunk 12 above)
Comment (kibo-susantasamanta) [reply components/KiboSearchSuggestion.vue:117]: it's locally scoped using class.

File: composables/useSearchSuggestions.ts (hunk 14 above)
Comment (kibo-susantasamanta) [reply composables/useSearchSuggestions.ts]: done.

File: components/KiboSearchSuggestion.vue (hunk 10 above)
Comment (kibo-susantasamanta) [reply components/KiboSearchSuggestion.vue:19]: Fixed with rem.

File: composables/useSearchSuggestions.ts (hunk 13 above)
Comment (kibo-susantasamanta) [reply composables/useSearchSuggestions.ts:24]: done

File: components/KiboSearchBar.vue (hunk 6 above)
Comment (kibo-susantasamanta) [reply components/KiboSearchBar.vue]: used.resolved

File: components/KiboSearchBar.vue (hunk 8 above)
Comment (kibo-susantasamanta) [reply components/KiboSearchBar.vue]: resolved

**Pull request merged**